docs/_guide/03-template-reference.md

```diff
@@ -254,44 +254,126 @@ Directives are functions that can extend lit-html by customizing the way a bindi
+  Count: <span>${asyncReplace(countUp())}</span>.
+`, document.body);
+```
+
+In the near future, `ReadableStream`s will be async iterables, enabling streaming `fetch()` directly into a template:
+
+```javascript
+// Endpoint that returns a billion digits of PI, streamed.
+const url =
+    'https://cors-anywhere.herokuapp.com/http://stuff.mit.edu/afs/sipb/contrib/pi/pi-billion.txt';
+
+const streamingResponse = (async () => {
+  const response = await fetch(url);
+  return response.body.getReader();
+})();
+render(html`π is: ${asyncAppend(streamingResponse)}`, document.body);
+```
+
+### cache
+
+`cache(conditionalTemplate)`
+
+Location: text bindings
+
+Caches the rendered DOM nodes for templates when they're not in use. The `conditionalTemplate` argument is an expression that can return one of several templates. `cache` renders the current
+value of `conditionalTemplate`. When the template changes, the directive caches the _current_ DOM nodes before switching to the new value. 
 
 Example:
 
 ```js
-import { repeat } from 'lit-html/directives/repeat';
+const detailView = (data) => html`<div>...</div>`; 
+const summaryView = (data) => html`<div>...</div>`;
 
-const myTemplate = () => html`
-  <ul>
-    ${repeat(items, (i) => i.id, (i, index) => html`
-      <li>${index}: ${i.name}</li>`)}
-  </ul>
-`;
+html`${cache(data.showDetails
+  ? detailView(data) 
+  : summaryView(data)
+)}`
+```
+
+When lit-html re-renders a template, it only updates the modified portions: it doesn't create or remove any more DOM than it needs to. But when you switch from one template to another, lit-html needs to remove the old DOM and render a new DOM tree. 
+
+The `cache` directive caches the generated DOM for a given binding and input template. In the example above, it would cache the DOM for both the  `summaryView` and `detailView` templates. When you switch from one view to another, lit-html just needs to swap in the cached version of the new view, and and update it with the latest data.
+
+### classMap
+
+`class=${classMap(classObj)}`
+
+Location: attribute bindings (must be the entire value of the `class` attribute)
+
+Sets a list of classes based on an object. Each key in the object is treated as a class name, if the value associated with the key is truthy, that class is added to the element. 
+
+```
+let classes = { highlight: true, enabled: true, hidden: false };`
+
+html`<div class=${classMap(classes)>Classy text</div>`;
+// renders as <div class="highlight enabled">Classy text</div>
+```
+
+Note that you can only use `classMap` in an attribute binding for the `class` attribute, and it must be the entire value of the attribute.
+
+
+```
+// DON'T DO THIS
+html`<div class="someClass ${classMap(moreClasses}">Broken div</div>`;
 ```
 
-If no `keyFn` is provided, `repeat` will perform similar to a simple map of
-items to values, and DOM will be reused against potentially different items.
 
 ### ifDefined
 
 `ifDefined(value)`
 
+Location: attribute bindings
```

Comment:
might be nice to update the example to show that this _can_ be used as a partial value of an attribute. It's pretty nice for URLs, so maybe something like `<img src="/images/${ifDefined(image.filename)}">`